### PR TITLE
Add theming

### PR DESCRIPTION
**Pull request by pdlla**
[replaces https://github.com/reflex-frp/reflex-vty/pull/42]

Add Theming so that all widgets can be themed via a Reader monad

### CODE REVIEW

File: src/Reflex/Vty/Widget/Input.hs [hunk 1]
```diff
@@ -115,6 +116,7 @@ checkboxStyleTick = CheckboxStyle
 -- | Configuration options for a checkbox
 data CheckboxConfig t = CheckboxConfig
   { _checkboxConfig_checkboxStyle :: Behavior t CheckboxStyle
+  -- TODO DELETE and use HasTheme instead
```
Comment (pdlla) [inline src/Reflex/Vty/Widget/Input.hs:119]: will be done in a followup PR

File: src/Reflex/Vty/Widget/Text.hs [hunk 2]
```diff
@@ -33,9 +35,11 @@ data RichTextConfig t = RichTextConfig
 instance Reflex t => Default (RichTextConfig t) where
   def = RichTextConfig $ pure V.defAttr
 
+
+-- TODO delete this and use new local theming
```
Comment (pdlla) [inline src/Reflex/Vty/Widget/Text.hs:39]: will be done in a followup PR

or should we keep this for compatibility?

File: src/Reflex/Vty/Widget/Input/Text.hs [hunk 3]
```diff
@@ -59,12 +61,20 @@ textInput cfg = do
             goToDisplayLinePosition mx (st + my) dl
         ]
       click <- mouseDown V.BLeft
-      let cursorAttrs = ffor f $ \x -> if x then cursorAttributes else V.defAttr
-      let rows = (\w s c -> displayLines w V.defAttr c s)
+
+      -- TODO reverseVideo is prob not what we want. Does not work with `darkTheme` in example.hs (cursor is dark rather than light bg)
```
Comment (pdlla) [inline src/Reflex/Vty/Widget/Input/Text.hs:65]: Any suggestions for how to reverse the color in a V.Attr for cursor?

**Comment by pdlla** [comment]
Could someone review please? I'd love to get this PR merged.

**Comment by pdlla** [comment]
Can someone review? Is the design lacking in anyway?

**Comment by ali-abrar** [comment]
Thanks @pdlla

**Pull request merged**